GW2EIUpdater/Updater.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System.Diagnostics;
+using System.IO.Compression;
+using System.Reflection;
+using System.Text.Json;
+
+namespace GW2EIUpdater;
+
+public static class Updater
+{
+    /// <summary>
+    /// Structure to store the update information.
+    /// </summary>
+    public struct UpdateInfo(string current, string latest, string release, string size, string file, bool update)
+    {
+        /// <summary>
+        /// Current Elite Insights running version.
+        /// </summary>
+        public string CurrentVersion = current;
+        /// <summary>
+        /// Latest Elite Insights version found.
+        /// </summary>
+        public string LatestVersion = latest;
+        /// <summary>
+        /// Latest Elite Insights release page link.
+        /// </summary>
+        public string ReleasePageURL = release;
+        /// <summary>
+        /// Size of the downloadable file.
+        /// </summary>
+        public string DownloadSize = size;
+        /// <summary>
+        /// Name of the file to download.
+        /// </summary>
+        public string FileName = file;
+        /// <summary>
+        /// Wether a new Elite Insights update has been found or not.
+        /// </summary>
+        public bool UpdateAvailable = update;
+    }
+
+    /// <summary>
+    /// Elite Insights temporary folder name in the system 'Temp' folder.<br></br>
+    /// 'GW2EIUpdateTemp' for the UI and 'GW2EICLIUpdateTemp' for the CLI.
+    /// </summary>
+    public static string TempFolderName { get; private set; } = string.Empty;
+    
+    private static readonly bool _downloadCLI = false;
```

Comment:
This does not make much sense, it should be an argument for DownloadAndUpdate method or deduced from file name like in CheckForUpdate

Reply:
Yes I forgot to remove this one when i introduced isCli through file name